pkg/kubelet/util.go

```diff
@@ -17,103 +17,19 @@ limitations under the License.
 package kubelet
 
 import (
-	"fmt"
 	"os"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/kubernetes/pkg/capabilities"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
-	"k8s.io/kubernetes/pkg/securitycontext"
 )
 
 // Check whether we have the capabilities to run the specified pod.
 func canRunPod(pod *v1.Pod) error {
-	if !capabilities.Get().AllowPrivileged {
-		for _, container := range pod.Spec.Containers {
-			if securitycontext.HasPrivilegedRequest(&container) {
-				return fmt.Errorf("pod with UID %q specified privileged container, but is disallowed", pod.UID)
-			}
-		}
-		for _, container := range pod.Spec.InitContainers {
-			if securitycontext.HasPrivilegedRequest(&container) {
-				return fmt.Errorf("pod with UID %q specified privileged init container, but is disallowed", pod.UID)
-			}
-		}
-	}
-
-	if pod.Spec.HostNetwork {
-		allowed, err := allowHostNetwork(pod)
-		if err != nil {
-			return err
-		}
-		if !allowed {
-			return fmt.Errorf("pod with UID %q specified host networking, but is disallowed", pod.UID)
-		}
-	}
-
-	if pod.Spec.HostPID {
-		allowed, err := allowHostPID(pod)
-		if err != nil {
-			return err
-		}
-		if !allowed {
-			return fmt.Errorf("pod with UID %q specified host PID, but is disallowed", pod.UID)
-		}
-	}
-
-	if pod.Spec.HostIPC {
-		allowed, err := allowHostIPC(pod)
-		if err != nil {
-			return err
-		}
-		if !allowed {
-			return fmt.Errorf("pod with UID %q specified host ipc, but is disallowed", pod.UID)
-		}
-	}
-
-	return nil
-}
-
-// Determined whether the specified pod is allowed to use host networking
-func allowHostNetwork(pod *v1.Pod) (bool, error) {
-	podSource, err := kubetypes.GetPodSource(pod)
+	_, err := kubetypes.GetPodSource(pod)
```

Comment:
Won't this prevent us from running any pods that don't have the `kubernetes.io/config.source` annotation?

Reply:
from `k8s.io/kubernetes/pkg/kubelet/types/pod_update.go`:
```
// GetPodSource returns the source of the pod based on the annotation.
func GetPodSource(pod *v1.Pod) (string, error) {
	if pod.Annotations != nil {
		if source, ok := pod.Annotations[ConfigSourceAnnotationKey]; ok {
			return source, nil
		}
	}
	return "", fmt.Errorf("cannot get source of pod %q", pod.UID)
}
```
So it looks like we'll get errors for Pods that lack that annotation. Also potentially for Pods with no annotations.
@tallclair

Reply:
After this PR, we'd want things to behave as if `AllowPrivileged` were set to `true`.

Reply:
fixed